src/crimson/osd/ops_executer.cc

```diff
@@ -483,184 +483,197 @@ OpsExecuter::execute_op(OSDOp& osd_op)
-    }, true);
-  case CEPH_OSD_OP_WRITEFULL:
-    return do_write_op([this, &osd_op] (auto& backend, auto& os, auto& txn) {
-      return backend.writefull(os, osd_op, txn, *osd_op_params, delta_stats);
-    }, true);
-  case CEPH_OSD_OP_APPEND:
-    return do_write_op([this, &osd_op] (auto& backend, auto& os, auto& txn) {
-      return backend.append(os, osd_op, txn, *osd_op_params, delta_stats);
-    }, true);
-  case CEPH_OSD_OP_TRUNCATE:
-    return do_write_op([this, &osd_op] (auto& backend, auto& os, auto& txn) {
-      // FIXME: rework needed. Move this out to do_write_op(), introduce
-      // do_write_op_no_user_modify()...
-      return backend.truncate(os, osd_op, txn, *osd_op_params, delta_stats);
-    }, true);
-  case CEPH_OSD_OP_ZERO:
-    return do_write_op([this, &osd_op] (auto& backend, auto& os, auto& txn) {
-      return backend.zero(os, osd_op, txn, *osd_op_params, delta_stats);
-    }, true);
-  case CEPH_OSD_OP_SETALLOCHINT:
-    return osd_op_errorator::now();
-  case CEPH_OSD_OP_SETXATTR:
-    return do_write_op([this, &osd_op] (auto& backend, auto& os, auto& txn) {
-      return backend.setxattr(os, osd_op, txn, delta_stats);
-    }, true);
-  case CEPH_OSD_OP_DELETE:
-    return do_write_op([this] (auto& backend, auto& os, auto& txn) {
-      return backend.remove(os, txn, delta_stats);
-    }, true);
-  case CEPH_OSD_OP_CALL:
-    return this->do_op_call(osd_op);
-  case CEPH_OSD_OP_STAT:
-    // note: stat does not require RD
-    return do_const_op([this, &osd_op] (/* const */auto& backend, const auto& os) {
-      return backend.stat(os, osd_op, delta_stats);
-    });
-  case CEPH_OSD_OP_TMAPUP:
-    // TODO: there was an effort to kill TMAP in ceph-osd. According to
-    // @dzafman this isn't possible yet. Maybe it could be accomplished
-    // before crimson's readiness and we'd luckily don't need to carry.
-    return dont_do_legacy_op();
-
-  // OMAP
-  case CEPH_OSD_OP_OMAPGETKEYS:
-    return do_read_op([this, &osd_op] (auto& backend, const auto& os) {
-      return backend.omap_get_keys(os, osd_op, delta_stats);
-    });
-  case CEPH_OSD_OP_OMAPGETVALS:
-    return do_read_op([this, &osd_op] (auto& backend, const auto& os) {
-      return backend.omap_get_vals(os, osd_op, delta_stats);
-    });
-  case CEPH_OSD_OP_OMAP_CMP:
-    return  do_read_op([this, &osd_op] (auto& backend, const auto& os) {
-      return backend.omap_cmp(os, osd_op, delta_stats);
-    });
-  case CEPH_OSD_OP_OMAPGETHEADER:
-    return do_read_op([this, &osd_op] (auto& backend, const auto& os) {
-      return backend.omap_get_header(os, osd_op, delta_stats);
-    });
-  case CEPH_OSD_OP_OMAPGETVALSBYKEYS:
-    return do_read_op([this, &osd_op] (auto& backend, const auto& os) {
-      return backend.omap_get_vals_by_keys(os, osd_op, delta_stats);
-    });
-  case CEPH_OSD_OP_OMAPSETVALS:
+  return osd_op_ierrorator::now().safe_then_interruptible([this, &osd_op]()
```

Comment:
A lot o code shuffling. IIUC the intention here is handle all the returns in one go. If so, we could rename the method into e.g. `do_execute_op()` and define a new `execute_op()` which would: 1) call `do_execute_op()` and handle the error.